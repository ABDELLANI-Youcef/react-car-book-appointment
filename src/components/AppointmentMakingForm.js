import { useState } from 'react';
import PropTypes from 'prop-types';
import styles from '../styles/Form.module.css';

const AppointmentMakingForm = ({ clickHandler }) => {
  const [dateCity, setDateCity] = useState({
    date: '',
    city: '',
  });

  const handleChange = (e) => {
    let { city, date } = dateCity;
    const { value } = e.target;
    switch (e.target.id) {
      case 'appointmentCity':
        city = value;
        break;

      default:
        date = value;
        break;
    }
    setDateCity({ city, date });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    clickHandler(dateCity);
  };
  return (
    <form onSubmit={handleSubmit}>
      <div className={styles.div_label}>

        <label htmlFor="appointmentCity" className={styles.input_line}>
          the city:
          <input id="appointmentCity" type="text" value={dateCity.city} onChange={handleChange} />
        </label>
      </div>
      <div className={styles.div_label}>
        <label htmlFor="appointmentDate" className={styles.input_line}>
          date:
          <input id="appointmentDate" type="date" value={dateCity.date} onChange={handleChange} />
        </label>
      </div>
      <div className={styles.div_label}>
        <input type="submit" value="Submit" className={styles.submit} />
      </div>
    </form>
  );
};

AppointmentMakingForm.propTypes = {
  clickHandler: PropTypes.func.isRequired,
};

export default AppointmentMakingForm;
