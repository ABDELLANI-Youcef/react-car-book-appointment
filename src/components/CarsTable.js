import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import styles from '../styles/Home.module.css';

const CarsTable = ({ cars, admin, clickHandler }) => {
  const handleClick = (e) => {
    const carId = parseInt(e.target.dataset.carId, 10);
    clickHandler(carId);
  };

  const k = cars.map((e) => {
    let deleteTD = null;
    let editTD = null;
    if (admin) {
      deleteTD = (<button type="button" data-car-id={e.id} onClick={handleClick}>Delete</button>);
      editTD = (<Link to={{ pathname: '/edit_car', state: { car: e } }}>edit the car</Link>);
    }
    return (
      <div key={e.id} className={styles.product_item}>
        <Link to={{ pathname: '/make-appointment', state: { car: e } }}>
          <img src={`https://youcef-cars-book-appointment.herokuapp.com${e.image}`} alt={`${e.mark} ${e.model}`} className={styles.product_image} />
        </Link>
        <div>
          <div className={styles.product_detail}>
            <span className={styles.detail_entry}>Mark:</span>
            {e.mark}
          </div>
          <div className={styles.product_detail}>
            <span className={styles.detail_entry}>Model:</span>
            {e.model}
          </div>
          <div className={styles.product_detail}>
            <span className={styles.detail_entry}>Year:</span>
            {e.year}
          </div>
          <div className={styles.product_detail}>
            {deleteTD}
            {editTD}
          </div>
        </div>
      </div>
    );
  });

  return (
    <div>
      { k }
    </div>
  );
};

CarsTable.propTypes = {
  admin: PropTypes.bool.isRequired,
  cars: PropTypes.arrayOf(PropTypes.any).isRequired,
  clickHandler: PropTypes.func.isRequired,
};

export default CarsTable;
