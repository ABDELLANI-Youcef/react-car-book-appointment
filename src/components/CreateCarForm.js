import { useState } from 'react';
import PropTypes from 'prop-types';
import styles from '../styles/Form.module.css';

const CreateCarForm = ({ clickHandler }) => {
  const [carData, setCarData] = useState({
    mark: '',
    model: '',
    year: 2021,
    price: 0,
    image: '',
    imageFile: '',
  });

  const handleChange = (e) => {
    const { id } = e.target;
    const { value } = e.target;
    const { files } = e.target;
    let {
      mark, model, year, image, imageFile, price,
    } = carData;
    switch (id) {
      case 'mark_input':
        mark = value;
        break;

      case 'model_input':
        model = value;
        break;

      case 'image_input':
        image = value;
        [imageFile] = files;
        break;

      case 'year_input':
        year = parseInt(value, 10);
        break;

      default:
        price = parseInt(value, 10);
        break;
    }

    setCarData({
      mark, model, year, image, imageFile, price,
    });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    clickHandler(carData);
  };

  return (
    <form onSubmit={handleSubmit} className={styles.form_div}>
      <div className={styles.div_label}>
        <label htmlFor="mark_input" className={styles.input_line}>
          Mark:
          {' '}
          <input id="mark_input" type="text" value={carData.mark} onChange={handleChange} />
        </label>
      </div>
      <div className={styles.div_label}>
        <label htmlFor="model_input" className={styles.input_line}>
          Model:
          {' '}
          <input id="model_input" type="text" value={carData.model} onChange={handleChange} />
        </label>
      </div>
      <div className={styles.div_label}>
        <label htmlFor="year_input" className={styles.input_line}>
          Year:
          {' '}
          <input id="year_input" type="number" value={carData.year} onChange={handleChange} />
        </label>
      </div>
      <div className={styles.div_label}>
        <label htmlFor="price_input" className={styles.input_line}>
          Price:
          {' '}
          <input id="price_input" type="number" value={carData.price} onChange={handleChange} />
        </label>
      </div>
      <div className={styles.div_label}>
        <label htmlFor="image_input" className={styles.input_line}>
          Car image:
          {' '}
          <input id="image_input" type="file" value={carData.image} onChange={handleChange} />
        </label>
      </div>
      <input type="submit" value="submit" className={styles.submit} />
    </form>
  );
};

CreateCarForm.propTypes = {
  clickHandler: PropTypes.func.isRequired,
};

export default CreateCarForm;
