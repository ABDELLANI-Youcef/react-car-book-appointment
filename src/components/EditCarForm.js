import { useState } from 'react';
import PropTypes from 'prop-types';
import styles from '../styles/Form.module.css';

const EditCarForm = ({ car, clickHandler }) => {
  const [cardata, setCardata] = useState({
    mark: car.mark,
    model: car.model,
    year: car.year,
    price: car.price,
  });
  const handleChange = (e) => {
    const { id } = e.target;
    const { value } = e.target;
    let {
      mark, model, year, price,
    } = cardata;
    switch (id) {
      case 'mark_edit':
        mark = value;
        break;

      case 'model_edit':
        model = value;
        break;

      case 'year_edit':
        year = parseInt(value, 10);
        break;

      default:
        price = parseInt(value, 10);
        break;
    }
    setCardata({
      mark, model, year, price,
    });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    clickHandler(cardata);
  };

  return (
    <form onSubmit={handleSubmit} className={styles.form_div}>
      <div className={styles.div_label}>
        <label htmlFor="mark_edit" className={styles.input_line}>
          mark:
          {' '}
          <input id="mark_edit" type="text" value={cardata.mark} onChange={handleChange} />
        </label>
      </div>
      <div className={styles.div_label}>
        <label htmlFor="model_edit" className={styles.input_line}>
          model:
          {' '}
          <input id="model_edit" type="text" value={cardata.model} onChange={handleChange} />
        </label>
      </div>
      <div className={styles.div_label}>
        <label htmlFor="year_edit" className={styles.input_line}>
          year:
          {' '}
          <input id="year_edit" type="number" value={cardata.year} onChange={handleChange} />
        </label>
      </div>
      <div className={styles.div_label}>
        <label htmlFor="price_edit" className={styles.input_line}>
          Price:
          {' '}
          <input id="price_edit" type="number" value={cardata.price} onChange={handleChange} />
        </label>
      </div>
      <input type="submit" value="submit" className={styles.submit} />
    </form>
  );
};

EditCarForm.propTypes = {
  car: PropTypes.objectOf(PropTypes.any).isRequired,
  clickHandler: PropTypes.func.isRequired,
};

export default EditCarForm;
