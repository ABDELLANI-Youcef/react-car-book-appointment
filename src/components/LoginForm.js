import { useState } from 'react';
import PropTypes from 'prop-types';

const LoginForm = ({ handleClick }) => {
  const [data, setData] = useState({
    email: '',
    password: '',
  });

  const handleChange = (e) => {
    let {
      email, password,
    } = data;
    switch (e.target.id) {
      case 'email':
        email = e.target.value;
        break;

      default:
        password = e.target.value;
        break;
    }
    setData({
      email, password,
    });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    handleClick(data);
  };
  return (
    <form onSubmit={handleSubmit}>
      <label htmlFor="email">
        E-mail:
        {' '}
        <input id="email" type="email" value={data.email} onChange={handleChange} />
      </label>
      <label htmlFor="password">
        Password:
        {' '}
        <input id="password" type="password" value={data.password} onChange={handleChange} />
      </label>

      <input type="submit" value="submit" />
    </form>
  );
};

LoginForm.propTypes = {
  handleClick: PropTypes.func.isRequired,
};

export default LoginForm;
