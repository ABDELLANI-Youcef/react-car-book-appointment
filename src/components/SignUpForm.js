import { useState } from 'react';
import PropTypes from 'prop-types';
import styles from '../styles/Form.module.css';

const SignUpForm = ({ handleClick }) => {
  const [data, setData] = useState({
    username: '',
    email: '',
    password: '',
    confirmPassword: '',
  });

  const handleChange = (e) => {
    let {
      username, email, password, confirmPassword,
    } = data;
    switch (e.target.id) {
      case 'username':
        username = e.target.value;
        break;

      case 'email':
        email = e.target.value;
        break;

      case 'password':
        password = e.target.value;
        break;

      default:
        confirmPassword = e.target.value;
        break;
    }
    setData({
      username, email, password, confirmPassword,
    });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    handleClick(data);
  };
  return (
    <form onSubmit={handleSubmit} className={styles.form_div}>
      <div className={styles.div_label}>
        <label htmlFor="username" className={styles.input_line}>
          Username:
          {' '}
          <input id="username" type="text" value={data.username} onChange={handleChange} />
        </label>
      </div>
      <div className={styles.div_label}>
        <label htmlFor="email" className={styles.input_line}>
          E-mail:
          {' '}
          <input id="email" type="email" value={data.email} onChange={handleChange} />
        </label>
      </div>
      <div className={styles.div_label}>
        <label htmlFor="password" className={styles.input_line}>
          Password:
          {' '}
          <input id="password" type="password" value={data.password} onChange={handleChange} />
        </label>
      </div>
      <div className={styles.div_label}>
        <label htmlFor="confirmPassword" className={styles.input_line}>
          Confirm password:
          {' '}
          <input id="confirmPassword" type="password" value={data.confirmPassword} onChange={handleChange} />
        </label>
      </div>
      <input type="submit" value="submit" className={styles.submit} />
    </form>
  );
};

SignUpForm.propTypes = {
  handleClick: PropTypes.func.isRequired,
};

export default SignUpForm;
