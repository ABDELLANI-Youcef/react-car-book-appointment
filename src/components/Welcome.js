import styles from '../styles/Welcome.module.css';

const Welcome = () => (
  <div className={styles.container}>
    <p>
      Welcome in our site. In this application we simulate
      the process to book appointments of cars.
    </p>
    <p>
      Please sign up or log in if you have already an account to use this plateform.
    </p>
  </div>
);

export default Welcome;
