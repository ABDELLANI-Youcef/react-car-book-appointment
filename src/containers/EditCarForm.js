import { useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

const sendEditCar = async (auth, cardata, carId) => {
  try {
    const body = {
      mark: cardata.mark,
      model: cardata.model,
      year: cardata.year,
      price: cardata.price,
    };
    const options = {
      method: 'PUT',
      headers: {
        Accept: 'application/json',
        'Content-Type': 'application/json',
        Authorization: auth.authToken,
      },
      body: JSON.stringify(body),
    };
    await fetch(`http://[::1]:3000/cars/${carId}`, options);
    return true;
  } catch (error) {
    return false;
  }
};

const EditCarForm = ({ location, auth }) => {
  const { car } = location.state;
  const [cardata, setCardata] = useState({
    mark: car.mark,
    model: car.model,
    year: car.year,
    price: car.price,
  });

  const handleChange = (e) => {
    const { id } = e.target;
    const { value } = e.target;
    let {
      mark, model, year, price,
    } = cardata;
    switch (id) {
      case 'mark_edit':
        mark = value;
        break;

      case 'model_edit':
        model = value;
        break;

      case 'year_edit':
        year = parseInt(value, 10);
        break;

      default:
        price = parseInt(value, 10);
        break;
    }
    setCardata({
      mark, model, year, price,
    });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (auth.admin) {
      sendEditCar(auth, cardata, car.id);
    }
  };

  return (
    <div>
      <form onSubmit={handleSubmit}>
        <label htmlFor="mark_edit">
          mark:
          {' '}
          <input id="mark_edit" type="text" value={cardata.mark} onChange={handleChange} />
        </label>
        <label htmlFor="model_edit">
          model:
          {' '}
          <input id="model_edit" type="text" value={cardata.model} onChange={handleChange} />
        </label>
        <label htmlFor="year_edit">
          year:
          {' '}
          <input id="year_edit" type="number" value={cardata.year} onChange={handleChange} />
        </label>
        <label htmlFor="price_edit">
          Price:
          {' '}
          <input id="price_edit" type="number" value={cardata.price} onChange={handleChange} />
        </label>
        <input type="submit" value="submit" />
      </form>
    </div>
  );
};

EditCarForm.propTypes = {
  location: PropTypes.objectOf(PropTypes.any).isRequired,
  auth: PropTypes.objectOf(PropTypes.any).isRequired,
};

const mapStateToProp = (state) => ({
  auth: state.authentication,
});

export default connect(mapStateToProp)(EditCarForm);
