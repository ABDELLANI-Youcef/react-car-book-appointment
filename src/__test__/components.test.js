import renderer from 'react-test-renderer';
import {
  BrowserRouter,
} from 'react-router-dom';
import Welcome from '../components/Welcome';
import CreateCarForm from '../components/CreateCarForm';
import LoginForm from '../components/LoginForm';
import SignUpForm from '../components/SignUpForm';
import EditCarForm from '../components/EditCarForm';
import AppointmentMakingForm from '../components/AppointmentMakingForm';
import CarsTable from '../components/CarsTable';

it('should render the welcome page correctly', () => {
  const welcomePage = renderer.create(<Welcome />).toJSON;
  expect(welcomePage).toMatchSnapshot();
});

it('should render CreateCarForm correctly', () => {
  const createCarForm = renderer.create(<CreateCarForm clickHandler={(() => true)} />).toJSON();
  expect(createCarForm).toMatchSnapshot();
});

it('should render LoginForm correctly', () => {
  const loginForm = renderer.create(<LoginForm handleClick={(() => true)} />).toJSON();
  expect(loginForm).toMatchSnapshot();
});

it('should render SignUpForm correctly', () => {
  const signUpForm = renderer.create(<SignUpForm handleClick={(() => true)} />).toJSON();
  expect(signUpForm).toMatchSnapshot();
});

it('should render EditCarForm correctly', () => {
  const carObject = {
    id: 1,
    mark: 'Bugatti',
    model: 'Chiron',
    year: 2020,
    price: 500,
  };
  const editCarForm = renderer.create(<EditCarForm car={carObject} clickHandler={(() => true)} />)
    .toJSON();
  expect(editCarForm).toMatchSnapshot();
});

it('should render AppointmentMakingForm correctly', () => {
  const appointmentMakingForm = renderer.create(
    <AppointmentMakingForm clickHandler={(() => true)} />,
  ).toJSON();
  expect(appointmentMakingForm).toMatchSnapshot();
});

it('should render CarsTable correctly', () => {
  const carsList = [
    {
      id: 1,
      mark: 'Bugatti',
      model: 'Chiron',
      year: 2020,
      price: 500,
      image: 'img/test.png',
    },
  ];
  const carsTable = renderer.create(
    <BrowserRouter>
      <CarsTable admin cars={carsList} clickHandler={(() => true)} />
    </BrowserRouter>,
  ).toJSON();
  expect(carsTable).toMatchSnapshot();
});
