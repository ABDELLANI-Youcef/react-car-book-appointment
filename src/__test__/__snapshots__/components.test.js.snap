// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AppointmentMakingForm correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="div_label"
  >
    <label
      className="input_line"
      htmlFor="appointmentCity"
    >
      the city:
      <input
        id="appointmentCity"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
  </div>
  <div
    className="div_label"
  >
    <label
      className="input_line"
      htmlFor="appointmentDate"
    >
      date:
      <input
        id="appointmentDate"
        onChange={[Function]}
        type="date"
        value=""
      />
    </label>
  </div>
  <div
    className="div_label"
  >
    <input
      className="submit"
      type="submit"
      value="Submit"
    />
  </div>
</form>
`;

exports[`should render CarsTable correctly 1`] = `
<div
  className="product_list"
>
  <div
    className="product_item"
  >
    <a
      href="/make-appointment"
      onClick={[Function]}
    >
      <img
        alt="Bugatti Chiron"
        className="product_image"
        src="https://youcef-cars-book-appointment.herokuapp.comimg/test.png"
      />
    </a>
    <div
      className="product_informations"
    >
      <div
        className="product_detail"
      >
        <span
          className="detail_entry"
        >
          Mark:
        </span>
        Bugatti
      </div>
      <div
        className="product_detail"
      >
        <span
          className="detail_entry"
        >
          Model:
        </span>
        Chiron
      </div>
      <div
        className="product_detail"
      >
        <span
          className="detail_entry"
        >
          Year:
        </span>
        2020
      </div>
      <div
        className="product_detail"
      >
        <a
          href="/make-appointment"
          onClick={[Function]}
        >
          Reserve an appointment to try this car
        </a>
      </div>
      <div
        className="product_detail"
      >
        <button
          data-car-id={1}
          onClick={[Function]}
          type="button"
        >
          Delete
        </button>
        <a
          href="/edit_car"
          onClick={[Function]}
        >
          edit the car
        </a>
      </div>
    </div>
  </div>
</div>
`;

exports[`should render CreateCarForm correctly 1`] = `
<form
  className="form_div"
  onSubmit={[Function]}
>
  <div
    className="div_label"
  >
    <label
      className="input_line"
      htmlFor="mark_input"
    >
      Mark:
       
      <input
        id="mark_input"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
  </div>
  <div
    className="div_label"
  >
    <label
      className="input_line"
      htmlFor="model_input"
    >
      Model:
       
      <input
        id="model_input"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
  </div>
  <div
    className="div_label"
  >
    <label
      className="input_line"
      htmlFor="year_input"
    >
      Year:
       
      <input
        id="year_input"
        onChange={[Function]}
        type="number"
        value={2021}
      />
    </label>
  </div>
  <div
    className="div_label"
  >
    <label
      className="input_line"
      htmlFor="price_input"
    >
      Price:
       
      <input
        id="price_input"
        onChange={[Function]}
        type="number"
        value={0}
      />
    </label>
  </div>
  <div
    className="div_label"
  >
    <label
      className="input_line"
      htmlFor="image_input"
    >
      Car image:
       
      <input
        id="image_input"
        onChange={[Function]}
        type="file"
        value=""
      />
    </label>
  </div>
  <input
    className="submit"
    type="submit"
    value="submit"
  />
</form>
`;

exports[`should render EditCarForm correctly 1`] = `
<form
  className="form_div"
  onSubmit={[Function]}
>
  <div
    className="div_label"
  >
    <label
      className="input_line"
      htmlFor="mark_edit"
    >
      mark:
       
      <input
        id="mark_edit"
        onChange={[Function]}
        type="text"
        value="Bugatti"
      />
    </label>
  </div>
  <div
    className="div_label"
  >
    <label
      className="input_line"
      htmlFor="model_edit"
    >
      model:
       
      <input
        id="model_edit"
        onChange={[Function]}
        type="text"
        value="Chiron"
      />
    </label>
  </div>
  <div
    className="div_label"
  >
    <label
      className="input_line"
      htmlFor="year_edit"
    >
      year:
       
      <input
        id="year_edit"
        onChange={[Function]}
        type="number"
        value={2020}
      />
    </label>
  </div>
  <div
    className="div_label"
  >
    <label
      className="input_line"
      htmlFor="price_edit"
    >
      Price:
       
      <input
        id="price_edit"
        onChange={[Function]}
        type="number"
        value={500}
      />
    </label>
  </div>
  <input
    className="submit"
    type="submit"
    value="submit"
  />
</form>
`;

exports[`should render LoginForm correctly 1`] = `
<form
  className="form_div"
  onSubmit={[Function]}
>
  <div
    className="div_label"
  >
    <label
      className="input_line"
      htmlFor="email"
    >
      E-mail:
       
      <input
        id="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </label>
  </div>
  <div
    className="div_label"
  >
    <label
      className="input_line"
      htmlFor="password"
    >
      Password:
       
      <input
        id="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </label>
  </div>
  <input
    className="submit"
    type="submit"
    value="submit"
  />
</form>
`;

exports[`should render SignUpForm correctly 1`] = `
<form
  className="form_div"
  onSubmit={[Function]}
>
  <div
    className="div_label"
  >
    <label
      className="input_line"
      htmlFor="username"
    >
      Username:
       
      <input
        id="username"
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
  </div>
  <div
    className="div_label"
  >
    <label
      className="input_line"
      htmlFor="email"
    >
      E-mail:
       
      <input
        id="email"
        onChange={[Function]}
        type="email"
        value=""
      />
    </label>
  </div>
  <div
    className="div_label"
  >
    <label
      className="input_line"
      htmlFor="password"
    >
      Password:
       
      <input
        id="password"
        onChange={[Function]}
        type="password"
        value=""
      />
    </label>
  </div>
  <div
    className="div_label"
  >
    <label
      className="input_line"
      htmlFor="confirmPassword"
    >
      Confirm password:
       
      <input
        id="confirmPassword"
        onChange={[Function]}
        type="password"
        value=""
      />
    </label>
  </div>
  <input
    className="submit"
    type="submit"
    value="submit"
  />
</form>
`;

exports[`should render the welcome page correctly 1`] = `[Function]`;
